EIPS/eip-7867.md

```diff
@@ -0,0 +1,553 @@
+
+### RPC Interface
+
+The following subsections are modifications to the API endpoints from EIP-5792.
+
+If a request does not match the schema defined below, the wallet MUST reject the request with an error code of 0 <!-- TODO -->.
+
+#### `wallet_sendCalls`
+
+The following JSON Schema SHALL be inserted, in the request object, as values of either the batch-scope or call-scope `capabilities` objects (as appropriate) with a key of `flow-control`.
+
+##### Batch-scope
+
+###### Schema
+
+```json
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "type": "object",
+    "additionalProperties": false,
+    "properties": {
+      "optional": {
+        "type": "boolean"
+      },
+      "atomicity": {
+        "enum": ["strict", "loose", "none"]
+      }
+    }
+}
+```
+
+###### Example Request
+
+```js
+[
+  {
+    "version": "1.0",
+    "from": "0xd46e8dd67c5d32be8058bb8eb970870f07244567",
+    "chainId": "0x01",
+    "calls": [],
+    "capabilities": {
+      "flow-control": {
+        "atomicity": "loose"
+      }
+    }
+  }
+]
+
+```
+
+
+##### Call-scope
+
+###### Schema
+
+```json
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "type": "object",
+    "additionalProperties": false,
+    "properties": {
+      "optional": {
+        "type": "boolean"
+      },
+      "on-failure": {
+        "enum": ["rollback", "halt", "continue"]
+      }
+    }
+}
+```
+
+###### Example Request
+
+```js
+[
+  {
+    "version": "1.0",
+    "from": "0xd46e8dd67c5d32be8058bb8eb970870f07244567",
+    "chainId": "0x01",
+    "calls": [
+      {
+        "to": "0xd46e8dd67c5d32be8058bb8eb970870f07244567",
+        "value": "0x182183",
+        "data": "0xfbadbaf01",
+        "capabilities": {
+            "flow-control": {
+                "on-failure": "continue"
+            }
+        }
+      }
+    ]
+  }
+]
+```
+
+#### `wallet_getCapabilities`
+
+The following JSON Schema is inserted into the per-chain object returned from `wallet_getCapabilities` with a key of `flow-control`.
+
+##### Schema
+
+```json
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "type": "object",
+    "additionalProperties": false,
+    "minProperties": 1,
+    "properties": {
+        "none": { "$ref": "#/$defs/on-failure" },
+        "loose": { "$ref": "#/$defs/on-failure" },
+        "strict": { "$ref": "#/$defs/on-failure" }
+    },
+    "$defs": {
+        "on-failure": {
+            "type": "array",
+            "uniqueItems": true,
+            "minItems": 1,
+            "items": {
+                "enum": ["rollback", "halt", "continue"]
+            }
+        }
+    }
+}
+```
+
+##### Example Response
+
+```js
+{
+    "0x1": {
+        "flow-control": {
+            "loose": ["halt", "continue"],
+            "strict": ["continue"]
```

Comment:
Shouldn't this define `on-failure` flow control for `call-scope` and `atomicity` flow control for `batch-scope`?

Not sure how to interpret this part:
```
 "loose": ["halt", "continue"],
 "strict": ["continue"]
```

Reply:
This snippet is saying the wallet supports `on-failure` modes of `halt` and `continue` when using `loose` atomicity, but only `continue` when using `strict`.